src/platform_impl/windows/event_loop.rs

```diff
@@ -265,10 +265,37 @@ impl<T> EventLoopWindowTarget<T> {
     }
 }
 
+/// Returns the id of the main thread.
+///
+/// Windows has no real API to check if the current executing thread is the "main thread" unlike macOS.
+/// Unix-like OSes have some APIs or semantics that exist regarding whether the main thread is in use or
+/// if the main thread id equals the pid like on Linux.
+///
+/// Windows will let us look up the current thread's id but this leaves us with the question of what
+/// is the main thread's id?
+/// We would somehow need to get the main thread's id before a developer could spin off any other
+/// threads inside of the main entrypoint in order to emulate the capabilities of other platforms.
+///
+/// We can get the id of the main thread by using CRT initialization.
+/// CRT initialization can be used to setup global state within a program by calling a list of
+/// function pointers which should assign a value to a static field.
+/// To have the field's value set to the main thread id, we need to place our function pointer inside
+/// of the section `.CRT$XCU` so it is called before the main entrypoint.
+///
+/// Full details of CRT initialization can be found here:
+/// https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-initialization?view=msvc-160
 fn main_thread_id() -> DWORD {
     static mut MAIN_THREAD_ID: DWORD = 0;
+
+    /// Function pointer used in CRT initialization section to set the above static field's value.
```

Comment:
Does this doc comment actually work as a doc comment? As in: if you build the docs with `--document-private-items`, does this appear on `INIT_MAIN_THREAD_ID`?

Reply:
I don't even see `INIT_MAIN_THREAD_ID` when running with that option you mentioned.

Reply:
Does it show as a doc comment if you use your IDE's on-hover hint?

Reply:
yes 
![Screenshot_20210528_112624](https://user-images.githubusercontent.com/30619168/120014249-5b8d7780-bfd1-11eb-82df-f239103caa26.png)

Reply:
In that case, I'm fine with leaving it as-is.